### DISCUSSION TITLE
Difference Between Go and Rust SDKs for EigenLayer

### DISCUSSION THREAD

**Discussion by ihedbit** [Layr-Labs/eigensdk-rs#179] [Q&A] [answered]
Hi **EigenLayer** Team,

Thank you for your amazing work on this project. I noticed that you provide **SDKs** in both **Go** and **Rust**, which is great for developers working in different ecosystems.

I was wondering:
**1.  Are there any significant differences between the Go and Rust SDKs in terms of features, performance, or use cases?**
**2. Is one of them more actively maintained or recommended for specific types of projects?**

Looking forward to your insights!

Best regards,
**Hooman**

**Comment by supernovahs** [comment, marked as answer]
1) Goal is to have them feature ~ and both pass compliance tests with same input.
2) Both of them are actively maintained right now. You can use any of them depending on your pref.

**Reply by supernovahs** [reply]
Open to contributions if you have any feature improvement or bug fixes :)

**Reply by ihedbit** [reply]
From what I understand, the **Rust SDK is as feature-complete as the Go SDK.** Since I need to develop an AVS and the programming language isn't a constraint for me, can I assume that both SDKs are equivalent in terms of logic and functionality? 

I noticed that the Rust SDK consists of around 700 lines of code, whereas the Go SDK has approximately 6,000 lines. Considering this, and given my preference for performance, would it be better to opt for the Rust SDK, assuming it is indeed as complete as the Go SDK?

**Reply by supernovahs** [reply]
There is no benchmarks right now. But yeah, rust SDK is as complete as Go SDK
